core/src/main/java/com/datastrato/gravitino/storage/relational/service/CatalogMetaService.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.storage.relational.service;
+
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.EntityAlreadyExistsException;
+import com.datastrato.gravitino.HasIdentifier;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.exceptions.NoSuchEntityException;
+import com.datastrato.gravitino.meta.CatalogEntity;
+import com.datastrato.gravitino.storage.relational.mapper.CatalogMetaMapper;
+import com.datastrato.gravitino.storage.relational.mapper.MetalakeMetaMapper;
+import com.datastrato.gravitino.storage.relational.po.CatalogPO;
+import com.datastrato.gravitino.storage.relational.utils.POConverters;
+import com.datastrato.gravitino.storage.relational.utils.SessionUtils;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * The service class for catalog metadata. It provides the basic database operations for catalog.
+ */
+public class CatalogMetaService {
+  private static final CatalogMetaService INSTANCE = new CatalogMetaService();
+
+  public static CatalogMetaService getInstance() {
+    return INSTANCE;
+  }
+
+  private CatalogMetaService() {}
+
+  public CatalogEntity getCatalogByIdentifier(NameIdentifier identifier) {
+    checkCatalogNamespaceWithIdentifier(identifier);
```

Comment:
You can use `NameIdentifier.checkCatalog();` here.

Reply:
Done